.github/workflows/build-publish-goreleaser.yml

```diff
@@ -15,6 +15,8 @@ jobs:
     steps:
       - name: Checkout repository
         uses: actions/checkout@v4.2.1
+        with:
+          persist-credentials: false
```

Comment:
Is there a way to do this by default so that we don't have to spam it around everywhere? (and remember to add it in the future for new workflows?)

Reply:
afaik the only way to achieve this would be by wrapping actions/checkout in a composite GHA and this as the default.

Reply:
^ yeah this would be the only way

Reply:
Also, I don't think it matters *that much* it was an easy remedy to a lot of warning level errors.

I might try and add it to `gha-workflow-validator` so it will annotate these references to make sure there's persist credentials.